AdyenCard/Validators/CardValidationError.swift

```diff
@@ -0,0 +1,91 @@
+//
+// Copyright (c) 2024 Adyen N.V.
+//
+// This file is open source and available under the MIT license. See the LICENSE file for more info.
+//
+
+@_spi(AdyenInternal) import Adyen
+import Foundation
+
+internal enum CardValidationError: String, AnalyticsValidationError {
+    
+    /// Card number is not filled.
+    case cardNumberEmpty
+    
+    /// Card number is partially filled.
+    case cardNumberPartial
+    
+    /// Luhn check for the card bin failed.
+    case cardLuhnCheckFailed
+    
+    /// Card brand is not supported.
+    case cardUnsupported
+    
+    /// Expiry date is not filled.
+    case expiryDateEmpty
+    
+    /// Expiry date is partially filled.
+    case expiryDatePartial
+    
+    /// Card expired.
+    case cardExpired
+    
+    /// Expiry date too far into the future.
+    case expiryDateTooFar
+    
+    /// Security code is not filled.
+    case securityCodeEmpty
+    
+    /// Security code is partially filled.
+    case securityCodePartial
+    
+    internal var analyticsErrorCode: Int {
+        switch self {
+        case .cardNumberEmpty:
+            return 900
+        case .cardNumberPartial:
+            return 901
+        case .cardLuhnCheckFailed:
+            return 902
+        case .cardUnsupported:
+            return 903
+        case .expiryDateEmpty:
+            return 910
+        case .expiryDatePartial:
+            return 911
+        case .cardExpired:
+            return 912
+        case .expiryDateTooFar:
+            return 913
+        case .securityCodeEmpty:
+            return 920
+        case .securityCodePartial:
+            return 921
+        }
+    }
+    
+    internal var analyticsErrorMessage: String {
+        switch self {
+        case .cardNumberEmpty:
+            return "Empty card number field."
```

Comment:
Are we gonna localized this error messages?

Reply:
These messages are only meant for the backend and not to be shown to the user.